PicPi/AddPhotoViewController.swift

```diff
@@ -4,28 +4,51 @@
 //
 //  Created by Mustafa T. Mohammed on 12/15/20.
 //
+
 import Photos
 import UIKit
 import BSImagePicker
+import Promises
 
 class AddPhotoViewController: UIViewController {
-  
-  let sendButton = customBtnRoundCornerBlueWithShadow(type: .custom)
-  let photoUploadManager = PhotoUploadManager()
-  let doneButton = customBtnRoundCornerBlueWithShadow(type: .custom)
-  
+  private let photoViewModel: AddPhotoViewModel
+  private var registeredPictureFrames = [PictureFrame]()
+
+  private let sendButton = customBtnRoundCornerBlueWithShadow(type: .custom)
+  private let doneButton = customBtnRoundCornerBlueWithShadow(type: .custom)
   private let tableView = UITableView()
   private var margins: UILayoutGuide!
-  
+
+  init(selectedAssets: [PHAsset] = [], selectedPictureFrames: [PictureFrame] = []) {
+    self.photoViewModel = AddPhotoViewModel(
+      selectedAssets: selectedAssets,
+      selectedPictureFrames: selectedPictureFrames
+    )
+
+    super.init(nibName: nil, bundle: nil)
+  }
+
+  required init?(coder: NSCoder) {
+    fatalError("init(coder:) has not been implemented")
+  }
+
   override func loadView() {
     super.loadView()
+
     view.backgroundColor = .white
     margins = view.layoutMarginsGuide
     setupSendButton()
     setupDoneBtn()
     setupTableView()
-    photoUploadManager.delegate = self
-    
+  }
+
+  override func viewDidAppear(_ animated: Bool) {
+    super.viewDidAppear(animated)
+
+    injectPictureFrameManager().registeredFrames().then { [weak self] frames in
```

Comment:
Is it ok to use injectPictureFrameManager() directly like this?!

Reply:
Yee, could just make a let one line before then call registeredFrames but this works too :3